src/utils/csv.ts

```diff
@@ -0,0 +1,35 @@
+import { writeToString } from "@fast-csv/format";
+
+type RowInterface = {
+  [key: string]: string;
+};
+
+const getCSVRows = (rows: readonly RowInterface[]) => {
+  let results: RowInterface[] = [];
+  rows.map((row: RowInterface) => {
+    results.push(row);
+    return "";
+  });
+  return results;
+};
+
+export const downloadGeneratedCSVFile = async (
+  // Rows needed to be normal objects but as input type is WithId<User> I can not pass it to @fast-csv/format lib so I need to use any
+  // eslint-disable-next-line
+  rows: readonly any[],
+  headers: string[]
+) => {
```

Comment:
`WithId<User>` is just a 'fancy name' for an object with a specific shape. There should be no reason why it can't be passed through here. We never use `any`, and we basically never disable `eslint` rules. If you find yourself needing to do either, then your approach is probably wrong and there is a better way to go about it. If you don't know what that better way is, then please ask someone to help.

We don't use multiple positional parameters when calling functions, we use an object with explicitly named parameters.

The 'generate CSV file' and 'download CSV file' should be 2 separate functions, not combined together like this.


```suggestion

export interface DownloadGeneratedCSVFile {
  headers: string[]
  rows: readonly any[], // TODO: fix this with whatever the correct type is, but definitely don't use 'any'
}

export const downloadGeneratedCSVFile = async ({ rows, headers }: DownloadGeneratedCSVFile) => {
```

Reply:
fixed in a2c3ebfeb0bd6fc1d4c317f56172510fc58adcab